src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas2/ApplicationsController.kt

```diff
@@ -40,15 +40,28 @@ class ApplicationsController(
   private val userService: NomisUserService,
 ) : ApplicationsCas2Delegate {
 
-  override fun applicationsGet(isSubmitted: Boolean?, page: Int?): ResponseEntity<List<ApplicationSummary>> {
+  override fun applicationsGet(
+    isSubmitted: Boolean?,
+    page: Int?,
+    prisonCode: String?,
+  ): ResponseEntity<List<ApplicationSummary>> {
     val user = userService.getUserForRequest()
 
+    prisonCode?.let { if (prisonCode != user.activeCaseloadId) throw ForbiddenProblem() }
+
     val pageCriteria = PageCriteria("created_at", SortDirection.desc, page)
 
-    val (applications, metadata) = when (isSubmitted) {
-      true -> applicationService.getSubmittedApplicationsForUser(user, pageCriteria)
-      false -> applicationService.getUnsubmittedApplicationsForUser(user, pageCriteria)
-      null -> applicationService.getAllApplicationsForUser(user, pageCriteria)
+    val (applications, metadata) = when (prisonCode) {
```

Comment:
the service method could then be
```  fun getApplications(prisonCode: String?, isSubmitted: Boolean?, user: NomisUserEntity, pageCriteria: PageCriteria<String>) =
    when (prisonCode) {
      null -> when (isSubmitted) {
        true -> getSubmittedApplicationsForUser(user, pageCriteria)
        false -> getUnsubmittedApplicationsForUser(user, pageCriteria)
        null -> getAllApplicationsForUser(user, pageCriteria)
      }

      else -> when (isSubmitted) {
        true -> getSubmittedApplicationsByPrison(prisonCode, pageCriteria)
        false -> getUnsubmittedApplicationsByPrison(prisonCode, pageCriteria)
        null -> getAllApplicationsByPrison(prisonCode, pageCriteria)
      }
    }

```
It looks like there aren't unit tests for the new service methods introduced in this PR, so unit testing this would introduce some coverage for them too.

Reply:
Logic has been moved to ApplicationService now and some unit tests added to test the new service functions.